engine/lite/components/request-response/src/main/scala/pl/touk/nussknacker/engine/lite/components/requestresponse/jsonschema/common/sources/JsonSchemaRequestResponseSource.scala

```diff
@@ -74,13 +60,32 @@ class JsonSchemaRequestResponseSource(val nodeId: NodeId, schemaStr: String, met
     }
   })
 
-  protected def prepareSchema(rawSchema: JSONObject): Schema = {
-    SchemaLoader.builder()
-      .useDefaults(true)
-      .schemaJson(rawSchema)
-      .draftV7Support()
-      .build().load().build()
-      .asInstanceOf[Schema]
+  private def decodeJsonWithError(str: String): Json = CirceUtil.decodeJsonUnsafe[Json](str, "Provided json is not valid")
+
+  protected val limitDisplayedErrors: Int = 5
+
+  protected def prepareValidationErrorMessage(exception: Throwable): String = {
+    exception match {
+      case ve: ValidationException =>
+        val allMessages = ve.getAllMessages.asScala
+        val allMessagesSize = ve.getAllMessages.size()
+        if(allMessagesSize > limitDisplayedErrors) allMessages.take(limitDisplayedErrors)
```

Comment:
Why don'y you show all messages?

Reply:
The idea is to make it readable to the end user. To big message can be unreadable since it is going to be displayed in NU component.

Reply:
IMO BE should return all errors and FE should decide how many of them show in end result to user.. And it should be generic FE logic for all error messages.

Reply:
You've got a point. Removed limitation and now we are returning all messages.